synapse/handlers/space_summary.py

```diff
@@ -612,6 +618,17 @@ class _RoomQueueEntry:
     via: Sequence[str]
 
 
+@attr.s(frozen=True, slots=True, auto_attribs=True)
+class _RoomEntry:
+    room_id: str
+    # The room summary for this room.
+    room: JsonDict
+    # An iterable of the sorted, stripped children events for children of this room.
+    #
+    # This may not include all children.
+    children: List[JsonDict] = attr.ib(factory=list)
```

Comment:
wondering if this should be a Collection rater than necessarily a List (and the default could be the empty tuple rather than a new list).

Reply:
I originally had it as a `Collection`, this was causing issues though since in `get_space_summary` (for the local room case) we directly use it as `events`, but for the remote case we call `append()` on `events`.

I could probably restructure that code, but it seemed a bit silly to do that just to get a more generic type (when all the callers *do* actually set it to a list).

I can take another look if you'd like?

Reply:
Seems like it was easy enough actually!